src/main/scala/main/solutions/MazeSolver/Solver.scala

```diff
@@ -0,0 +1,33 @@
+package main.solutions.MazeSolver
+
+import scala.collection.mutable.{HashSet, Queue}
+
+object Solver {
+  def hasPathBFS(maze: Array[Array[Int]], start: Point, end: Point): Boolean = {
+    hasPathBFSHelper(maze, start, end, HashSet[Point]())
+  }
+  private def hasPathBFSHelper(maze: Array[Array[Int]], start: Point, end: Point, visited: HashSet[Point]): Boolean = {
+    val queue = Queue[Point]()
+    queue.enqueue(start)
+    while (queue.nonEmpty) {
```

Comment:
using "while" loops in Scala is not common, since it is more OOP/imperative style, while Scala is encouraging functional programming. the reason you are using "while here because you chose to use a mutable HashSet, which is also not the common pattern. The more common thing to do is to have to use recursion, and each time passing it a new instance of the set.

Reply:
Great, I will try to use immutable collections